sw/device/silicon_creator/rom/doc/rom_customization.md

```diff
@@ -0,0 +1,98 @@
+---
+Document: RFC
+Title: ROM Vendor Customizations
+Status: Draft
+Authors: Samuel Ortiz <sameo@opentitan.org>
+Reviewers:
+---
+
+# RFC: ROM Vendor Customizations 
+
+## Problem Statement
+
+Currently, OpenTitan mask ROM implementations lack customization options for silicon creators and vendors. While discrete OpenTitan ROMs can be directly built upon the [upstream implementation](https://github.com/lowRISC/opentitan/tree/master/sw/device/silicon_creator/rom), the same isn't true for integrated OpenTitan top ROMs.
+
+Integrated OpenTitan ROMs configure, initialize, and interact with external subsystems specific to the platform. These interactions often vary between vendors and platform variants. A single, fixed ROM implementation prevents downstream silicon creators and vendors from directly utilizing the upstream code, forcing them to fork the project and maintain custom versions internally.
+
+## Requirements
```

Comment:
Comment from @moidx in https://github.com/lowRISC/opentitan/issues/25633#issuecomment-2549678655:

Are we planning to define additional requirements for out of band verification? For example, can we ask silicon integrators to document the expected ROM version and associated hash for any given silicon release?

* The information can be used to perform out of band checks against the device attestation (e.g. it may help in providing additional supply chain hardening).
* The information can be classified as public or restricted depending on the silicon integration requirements.
* The information can be incorporated into OCP's S.A.F.E framework for integrations requiring third party audits; or by Common Criteria certification.

Reply:
> Are we planning to define additional requirements for out of band verification? For example, can we ask silicon integrators to document the expected ROM version and associated hash for any given silicon release?
> 
>     * The information can be used to perform out of band checks against the device attestation (e.g. it may help in providing additional supply chain hardening).

I am expecting a device attestation report to include its boot ROM hash, and that could be used by a verifier against a set of reference values that thus would need to include the expected ROM hash and version. Those reference values are typically built and signed by the silicon creators, so this is even more than just documenting.
Also, this should already be a requirement today, independent of this RFC proposal here.

Reply:
Comment from @moidx in https://github.com/lowRISC/opentitan/issues/25633#issuecomment-2551606844:

Hi @sameo, I brought this up because in cases where there are no customization hooks, the verifier can perform an out of band check by building the ROM from source. I understand that the out of band measurement is a requirement from the verifier's perspective, but thought it would be good to have this included in the documentation. Mentioning that this is covered by an attestation report is sufficient from my perspective.

Reply:
hey @moidx. I see, this makes sense.

I added an additional attestation requirement to address that one.